src/selectors/entities/channels.js

```diff
@@ -53,30 +54,39 @@ export {
     getMyCurrentChannelMembership,
 };
 
-export function getAllChannels(state) {
+import type {GlobalState} from '../../types/store';
+import type {Channel, ChannelStats, ChannelMembership} from '../../types/channels';
+import type {UsersState, UserProfile} from '../../types/users';
+import type {PreferenceType} from '../../types/preferences';
+import type {Post} from '../../types/posts';
+import type {TeamMembership} from '../../types/teams';
+
+type SortingType = 'recent' | 'alpha';
+
+export function getAllChannels(state: GlobalState): {[string]: Channel} {
```

Comment:
I like the RleationOneToOne and RleationOneToMany, but I don't like the RelationSelf, because it's is normally used to "Map" the objects by id, not to relate objects to itself. I would expect the exact same implementation but with a better name, like IDMappedObjects, and use `: IDMappedObjects<Channel>`.